scipio/examples/hello_world.rs

```diff
@@ -0,0 +1,44 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the
+// MIT/Apache-2.0 License, at your convenience
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2020 Datadog, Inc.
+//
+use futures::future::join_all;
+use scipio::{Local, LocalExecutor};
+use std::io::Result;
+
+async fn hello() {
+    let mut tasks = vec![];
+    for t in 0..5 {
+        tasks.push(Local::local(async move {
+            println!("{}: Hello {} ...", Local::id(), t);
+            Local::later().await;
+            println!("{}: ... {} World!", Local::id(), t);
+        }));
+    }
+    join_all(tasks).await;
+}
+
+fn main() -> Result<()> {
+    // There are two ways to create an executor, demonstrated in this example.
+    //
+    // We can create it in the current thread, and run it separately later...
+    let ex = LocalExecutor::new(Some(0))?;
+
+    // Or we can spawn a new thread with an executor inside.
+    let handle = LocalExecutor::spawn_executor("hello", Some(1), async move {
+        hello().await;
+    })?;
+
+    // If you create the executor manually, you have to run it like so.
+    //
+    // spawn_new() is the preferred way to create an executor!
+    ex.run(async move {
+        hello().await;
+    });
+
+    // The newly spawned executor runs on a thread, so we need to join on
+    // its handle so we can wait for it to finish
+    handle.join().unwrap();
```

Comment:
This is exactly why making main() -> std::io::Result instead of expect() was a bad idea:

```
  --> scipio/examples/hello_world.rs:42:18
   |
22 | fn main() -> Result<()> {
   |              ---------- expected `std::io::Error` because of this
...
42 |     handle.join()?;
   |                  ^ the trait `std::convert::From<std::boxed::Box<dyn std::any::Any + std::marker::Send>>` is not implemented for `std::io::Error`
```

now the code has inconsistent error handling so that even main return is useless because all other errors can't fit its type.

Reply:
btw, the fact Rust has many incompatible variants of Result<> type is painful. Maybe its worth having your own Result<> and return everywhere 1 type and make it compatible with at least 1 std (std::io ie).

Reply:
This is only a problem (and apparently one known to the rust community) with `.join()`
If you google this error message it should take you to places where the Rust folks tell you not to `?` on join.
(although now that I am searching again I have been so far unable).

You will get similar results if you spawn a thread manually and then `join` it, so I am not too concerned.

My understanding is that it would be possible to make this better and `?` on join if we implemented an error wrapper on `io::Error` that is send, but I am not too concerned about that either: if a thread fails in an unrecoverable way (at least for us), you are likely to want to panic the whole thing anyway: otherwise you are left with executors that are running while others have failed.

I feel like the example is good now: The user can see how to handle errors that happen at spawn time, and for asynchronous errors coming from threads we are recommending unwrap.

Reply:
I agree about the errors, but in practice it seems to me that everyone standardized on `io::Result` so as long as your errors are convertible to and from it, all should be good

Reply:
Hi Daniel - as we discussed, this won't work in practice because `JoinHandle` transforms the error into a boxed `std::any::Any` whose size is not known.

I don't know of a very good way to `?` from a join, but this is a problem that `thread::spawn` has already. We can fix the example when we find a good way (I'd like to), but tabling for now so we can make progress